### PR TITLE
Added parsing logic for the key=value body requests

### PR DESCRIPTION
**Pull request by KRoLer**
Added parsing logic for the requests of type **application/x-www-form-urlencoded** with the key=value body string. 

I didn't find another way to reach them except to extend an existing logic. 
If you think it will be useful I can update documentation for this feature. 
If you know another way how to do it I'll be also glad to hear it.

### CODE REVIEW

**Comment by tranhungt** [comment]
@KRoLer Thanks for this. I think this could be useful in extending the applicability of this transformer, since we should support multiple request body types.
I can look more into the wiremock library to see if there are any other possible solutions.
We should go ahead and add the documentation to this PR.

**Comment by Simon-Kaz** [comment]
Hey @KRoLer , I tried using your solution to parse an email, but I'm getting url encoded version back instead of the "@". Any way to disable the url encoding?

File: src/test/java/com/opentable/extension/BodyTransformerTest.java [hunk 1]
```diff
@@ -70,6 +70,58 @@ public void replaceNestedVariablesForXml() throws Exception {
...
+
+        wireMockRule.verify(postRequestedFor(urlEqualTo("/get/this")));
+
+    }
+	private void testKeyBodyRequest(String body) {
+		wireMockRule.stubFor(post(urlEqualTo("/get/this"))
+			.willReturn(aResponse()
+				.withStatus(200)
+				.withHeader("content-type", "application/json")
+				.withBody("{\"EmptyKey\":\"\"}")
```
Comment (tranhungt) [inline src/test/java/com/opentable/extension/BodyTransformerTest.java]: If this is returning empty string, then we'd never know that it tried to parse the "EmptyKey" url body?

File: src/main/java/com/opentable/extension/BodyTransformer.java [hunk 2]
```diff
@@ -54,7 +56,20 @@ public ResponseDefinition transform(Request request, ResponseDefinition response
...
+                //Validate is a body has the 'name=value' parameters
+                String body = request.getBodyAsString();
+                if(StringUtils.isNotEmpty(body) && (body.contains("&") || body.contains("=")))
+                {
+                    object = new HashMap();
+                    String [] pairedValues = request.getBodyAsString().split("&");
+                    for(String pair: pairedValues)
+                    {
+                        String[] values = pair.split("=");
+                        object.put(values[0], values.length > 1 ? values[1] : "");
```
Comment (tranhungt) [inline src/main/java/com/opentable/extension/BodyTransformer.java]: Shouldn't we `URLDecoder.decode` the values[1]?

**Comment by tranhungt** [comment]
@KRoLer Sorry for the late review. Could you check the comments, fix the conflict, and add some documentation? I can merge asap.

Thanks.

**Comment by KRoLer** [comment]
@tranhungt, thanks for the review. I've fixed issue for the case when it has only parameter name and commits it; I've not planned to change my PR. If you accept this solution at all, I'll refactor and prettify code before the merge. :-)

**Comment by KRoLer** [comment]
Hey, @tranhungt,

I've updated my PR. 
Now it had an actual code version, fixed test issues and added all needed logic, including UTF-8 values decoding.

**Comment by tranhungt** [comment]
Thanks @KRoLer this looks great! I'll check it out tomorrow morning.

**Pull request merged**